airbyte-ci/connectors/pipelines/pipelines/helpers/changelog.py

```diff
@@ -0,0 +1,117 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import datetime
+import re
+from operator import attrgetter
+from pathlib import Path
+
+import semver
+from typing_extensions import Set
+
+
+class Changelog:
+    def __init__(self, changelog_file_path: Path) -> None:
+        self.changelog_file_path = changelog_file_path
+        if not changelog_file_path.exists():
+            raise Exception(f"File {changelog_file_path} does not exist")
+        if not changelog_file_path.is_file():
+            raise Exception(f"File {changelog_file_path} is not a regular file")
+        self.original_markdown_lines = changelog_file_path.read_text().splitlines()
+        self.__parse_markdown()
+        self.new_entries: Set[ChangelogEntry] = set()
+
+    def add_entry(self, version: semver.Version, date: datetime.date, pull_request_number: int, comment: str) -> None:
+        self.new_entries.add(ChangelogEntry(date, version, pull_request_number, comment))
+
+    def to_markdown(self) -> str:
+        all_entries = set(self.old_entries.union(self.new_entries))
+        sorted_entries = sorted(
+            sorted(
+                sorted(sorted(all_entries, key=attrgetter("comment"), reverse=True), key=attrgetter("pr_number"), reverse=True),
+                key=attrgetter("date"),
+                reverse=True,
+            ),
+            key=attrgetter("version"),
+            reverse=True,
+        )
+        new_lines = (
+            self.original_markdown_lines[: self.changelog_entries_start_line_index]
+            + [line.to_str() for line in sorted_entries]
+            + self.original_markdown_lines[self.last_changelog_line_index + 1 :]
+        )
+        return "\n".join(new_lines) + "\n"
+
+    def __parse_markdown(self) -> None:
```

Comment:
Can we make this a public static or class method which would return a list of changelog entry? 
I'd find it more elegant to call it this way in the contrusctor:
`self.old_entries = self.parse_markdown(original_markdown_line)` 

Making it public and a static/class method can simplify testing and reuse in other contexts.